### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by fgsch**
Some refactor after removing the dynamic config.

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -543,7 +509,9 @@ func (f *File) Write(path string) (err error) {
 	if err != nil {
 		return fmt.Errorf("error creating config file: %w", err)
 	}
-	if err := toml.NewEncoder(fp).Encode(f); err != nil {
+	encoder := toml.NewEncoder(fp)
+	encoder.Indentation("")
```
Comment (Integralist) [inline pkg/config/config.go:513]: What problem is `.Indentation` solving? 

I'd like us to add a `// NOTE: ...` to clarify why we use it.

Comment (Integralist) [reply pkg/config/config.go:513]: UPDATE: I think I see why it's used in the `config_test.go` below.

**Review by Integralist — COMMENTED** [review]
LGTM. Couple of small comments.

File: pkg/config/config.go (hunk 1 above)
Comment (fgsch) [reply pkg/config/config.go:513]: Sure, I will add a comment.

Comment (fgsch) [reply pkg/config/config.go:513]: Done.

File: pkg/config/config.go [hunk 2]
```diff
@@ -484,20 +427,35 @@ func (f *File) MigrateLegacy() {
...
+
+		if verbose {
+			text.Output(out, `
+				Found your local configuration file (required to use the CLI) was using a legacy format.
+				File will be updated to the latest format.
+			`)
+			text.Break(out)
+		}
+		return true
+	case f.ConfigVersion != currentConfigVersion:
```
[the remark this answers is not in the document]
Comment (fgsch) [reply pkg/config/config.go:456]: That's fair but we currently don't have such a test. 

Do you want to cross that bridge when we get there or you would like to see a test for it?

File: pkg/config/config.go (hunk 1 above)
Comment (fgsch) [reply pkg/config/config.go:513]: To clarify the original question, this is removing the leading spaces. 
I'm not sure it qualifies as a problem here but I don't think it's needed either.

File: pkg/config/config.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (fgsch) [reply pkg/config/config.go]: Test added.

File: pkg/config/config.go [hunk 3]
```diff
@@ -347,7 +340,16 @@ func (f *File) Read(
...
 	verbose bool,
 ) error {
-	var useStatic bool
+	// Ensure the static config is sound. This should never happen (tm).
+	// We are checking this earlier to simplify the code later on.
+	var staticConfig File
+	err := toml.Unmarshal(Static, &staticConfig)
+	if err != nil {
+		errLog.Add(err)
+		return invalidConfigErr(err)
```
Comment (Integralist) [inline pkg/config/config.go:349]: The error message `ErrInvalidConfig` is a bit vague for a user who doesn't know anything about the CLI's internal config or the static embed etc, we should update the error and make it clearer to the user what we're referring to and in this scenario it's actually quite unexpected.

File: pkg/config/config.go [hunk 4]
```diff
@@ -43,30 +43,23 @@ const (
...
-	// UpdateSuccessful represents the message shown to a user when their
-	// application configuration has been updated successfully.
-	UpdateSuccessful = "Successfully updated platform compatibility and versioning information."
 )
 
-// ErrLegacyConfig indicates that the local configuration file is using the
-// legacy format.
-var ErrLegacyConfig = errors.New("the configuration file is in the legacy format")
+var (
+	CurrentConfigVersion int
```
Comment (Integralist) [inline pkg/config/config.go:49]: Let's get a doc string comment on this now it's public.

**Review by Integralist — APPROVED** [review]
Couple of minor comments, but otherwise approved.

File: pkg/config/config.go (hunk 3 above)
Comment (fgsch) [reply pkg/config/config.go:349]: Sure. Do you have any suggestions?

File: pkg/config/config.go (hunk 4 above)
Comment (fgsch) [reply pkg/config/config.go:49]: Added.

File: pkg/config/config.go (hunk 3 above)
Comment (fgsch) [reply pkg/config/config.go]: Updated.

**Review by Integralist — APPROVED** [review]
LGTM

**Pull request merged**